site/src/scaffold/scripts/popover.ts

```diff
@@ -0,0 +1,80 @@
+export function initPopovers(container: HTMLElement) {
+	container.addEventListener('click', (event: Event) => {
+		const allPopovers: HTMLDetailsElement[] = Array.from(
+			container.querySelectorAll('details.popover')
+		);
+
+		if (allPopovers.length === 0) {
+			return;
+		}
```

Comment:
I think the logic can be reduced by using the `<details>` element's toggle event and by removing the looping over all popovers, similar to the .dropdown handling.

```ts
container.addEventListener('toggle', event => {
  // Was a details.popover element toggled *open*?
  if (!(event.target instanceof HTMLDetailsElement) || !event.target.open || !event.target.classList.contains('popover')) {
    return;
  }
  
  const details = event.target;
  
  // Synchronize aria-expanded with the details element's open state.
  details.setAttribute('aria-expanded', details.open);

  // Make sure the details element collapses automatically...

  const checkTarget = (event: Event) => {
    if (!(event.target instanceof Element)) {
      return;
    }
    
    if (!details.contains(event.target) /*|| (event.type === 'click' && event.target.closest('.dropdown-menu-close')*/) {
      collapse();
      /*if (event.target.classList.contains('dropdown-menu-close')) {
        trigger.focus();
      }*/
    }
  };
  
  const keyHandler = (event: KeyboardEvent) => {
    if (event.keyCode === keyCodes.escape) {
      collapse();
      details.querySelector('summary')!.focus();
    }
  };
  
  const blurHandler = async () => {
    if ((document.activeElement?.nodeName ?? '') ==='IFRAME') {
      collapse();
    }
  };
  
  collapse = () => {
    container.removeEventListener('focus', checkTarget);
    container.removeEventListener('click', checkTarget);
    container.removeEventListener('touchstart', checkTarget);
    container.removeEventListener('keydown', keyHandler);
    window.removeEventListener('blur', blurHandler);
    details.removeEventListener('toggle', collapse);
    details.setAttribute('aria-expanded', 'false');
    details.toggleAttribute('open', false);
  };
  
  const useCapture = true;
  container.addEventListener('focus', checkTarget, useCapture);
  container.addEventListener('click', checkTarget);
  container.addEventListener('touchstart', checkTarget);
  container.addEventListener('keydown', keyHandler);
  window.addEventListener('blur', blurHandler);
  details.addEventListener('toggle', collapse);
});
```

Reply:
The idea is great, but you still have to assign it to each popover first. And because we can not guarantee that the popover component won't be added to the page sometime after the initial content loaded, we still need to keep the body's click event to verify the amount of popover's on a page. And every with every click we would loop through each popover and assign it the `toggleHandler` event, and to avoid tons of them assigned to the same popover we need to clean them up somehow too. So it doesn't really look like a path to follow, to be honest. Please correct me if I'm wrong :) 

The way I was also thinking about using the `toggle` event is it could be useful in switching the `aria-expanded` value, but it's the whole other story - when I added that attribute and was testing locally with Narrator and NVDA I found out that having this additional attribute doesn't make a difference in how the screen readers announce (does not actually) when the details component opens/closes. So I'm keeping that PR about event listeners, and want to work on accessibility in a separate one.

Reply:
Updated